src/vehicle_cmd.cpp

```diff
@@ -172,6 +172,10 @@ std::tuple<CommandCost, VehicleID, uint, uint16, CargoArray> CmdBuildVehicle(DoC
 		}
 
 		if (flags & DC_EXEC) {
+			if (type == VEH_TRAIN && use_free_vehicles && !(flags & DC_AUTOREPLACE)) { // check if the cars should be added to the new vehicle
```

Comment:
I know it is existing code, but please move that comment before the if-statement. Tnx :)

Reply:
It is a poor comment really, as the only "check" is do the parameters ask for it. So hopefully better comment on next line.